lib/ecto/adapters/clickhouse.ex

```diff
@@ -144,8 +146,30 @@ defmodule Ecto.Adapters.ClickHouse do
   defdelegate structure_dump(default, config), to: Ecto.Adapters.ClickHouse.Structure
 
   @impl Ecto.Adapter.Structure
-  def structure_load(_default, _config) do
-    raise "not implemented"
+  def structure_load(default, config) do
+    path = config[:dump_path] || Path.join(default, "structure.sql")
+
+    Logger.info("Loading database structure from file: #{path}")
```

Comment:
Is this needed? Do other adapter log it? I thought Ecto is responsible for doing logging for these mix tasks.

Reply:
I don't think ecto logs it. I think it is valuable for the developer to know from which path the structure was loaded

Reply:
Postgres adapter doesn't seem to log anything. I'd like the adapter to follow the Postgres one. That's what most people using Ecto are used to.

Reply:
Also Postgres seems to be using a command callback. I wonder if we can do the same. We'd call clickhouse-client with `--queries-file` pointing at the structure. It'd probably handle all edge cases that we right now are not thinking about.

Reply:
https://clickhouse.com/docs/en/integrations/sql-clients/cli

Reply:
I will remove it but update the error message when file reading files with the path of the read file

Reply:
ohh, okay. Can do that as well

Reply:
I've tried creating a file with the same schemas you are using the the tests, and then running `./clickhouse client --queries-file structure.sql`. Seems to have worked:

```sql
mac3.local :) show tables;

SHOW TABLES

Query id: aa93b6f7-b72e-43f2-9f86-d3796b492f9b

┌─name───────────────────────────────────────────┐
│ .inner_id.f88ef8c6-d634-4c72-830a-37981b840e1c │
│ candle_fragments                               │
│ candles_one_hour_amt                           │
│ ch_datetimes                                   │
│ ch_demo                                        │
│ ch_utf8                                        │
│ events                                         │
│ example                                        │
│ first_struture_load_table                      │
│ market_cope                                    │
│ second_struture_load_table                     │
└────────────────────────────────────────────────┘
```

Reply:
So yeah, I think this is how I'd like it to work. I can add `run_with_cmd` in a separate PR (since I think it's also a required callback) and then you'd be able to use it in this PR. Or you can do both, you decide :)

Reply:
pls see latest commit, I am now using clickhouse client

Reply:
it seems the clickhouse client is not available in the CI

Reply:
Should I just add a step which runs `curl https://clickhouse.com/ | sh` ?

Reply:
Yeah, let's try that, ecto_sql does something similar: https://github.com/elixir-ecto/ecto_sql/blob/master/Earthfile#L78

But they use package manager to download the client.

Reply:
ci failed, I will debug

Reply:
u can cancel the CI now @ruslandoga , think the issue is not being in the path

Reply:
@ruslandoga see latest fail https://github.com/plausible/ecto_ch/actions/runs/4828203615/jobs/8601742074

I think you have to enable something unless you know a different solution?

Reply:
I'll check.

Reply:
I wonder if we can add a mix task to install the client and add it to path. And then run it before the tests in CI. I'll try that out.

Reply:
`sudo ./clickhouse install` seems to work: https://github.com/plausible/ecto_ch/pull/61/files

Reply:
nice can you maybe merge your PR and then I will merge master into my branch or should I copy that?

Reply:
so I think this is now the only thing needed before we can merge it?

Reply:
I copied it! :)